### ISSUE TITLE
Unncessary semicolons in tutorials terminate statements, possible documentation improvement

### ISSUE THREAD

**Issue by brechtse** [tensorflow/tensorflow#57587]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Documentation Bug

### Source

source

### Tensorflow Version

2.9.1

### Custom Code

No

### Current Behaviour?

```shell
The documentation found at `https://www.tensorflow.org/guide/basics` under heading 'Training loops' as well as additional documentation on that tutorial/guide page have unnecessary semicolons (`;`) at end of statements, not necessary for Python syntactically. 

For example, `plt.legend();` under 'Training loops' heading throws warning errors in certain IDEs with AI-assisted code review such as VSCode because of the superfluous `;`.

I kindly recommend discontinuation of unnecessary semicolons and revision of this page to remove all that are not needed per proper syntax or convention. Kindly let me know if any more information is needed. Thank you for consideration.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import matplotlib
from matplotlib import pyplot as plt
matplotlib.rcParams['figure-figsize'] = [9, 6]

x = tf.linspace(-2, 2, 201)
x = tf.cast(x, tf.float32)

def f(x):
  y = x**2 + 2*x - 5
  return y

y = f(x) + tf.random.normal(shape=[201])

plt.plot(x.numpy(), y.numpy(), '.', label='Data')
plt.plot(x, f(x), label='Ground truth')
plt.legend();

_excerpted from: https://www.tensorflow.org/guide/basics_
```

### Relevant log output

```shell
n/a
```
</details>

**Comment by Indira-kumar** [comment]
Could you assign this issue to me?

**Comment by brechtse** [comment]
I think the issue was auto-assigned to @tilakrayal. If there is anything I can do to help facilitate and possibly get this resolved to help future learners, if indeed it deserves attention, kindly let me know. 

Best!
@brechtse

**Comment by tilakrayal** [comment]
@brechtse,
Semicolon used here in the doc is to denote separation rather than termination. It is intended and not able to find any issue while executing the code. Thank you!

**Comment by brechtse** [comment]
@tilakrayal thank you for your response. I understand the intent and also have noticed warning from IntelliCode in VSCode when running this, because of PEP8 style. Given the purpose and audience, however, perhaps such slight occurrence is not worth a PR or change. I am thankful for your review and will close this issue. closes #57587
